lib/web_ui/lib/src/engine/text_editing/text_editing.dart

```diff
@@ -849,6 +995,8 @@ abstract class DefaultTextEditingStrategy implements TextEditingStrategy {
   late InputConfiguration inputConfiguration;
   EditingState? lastEditingState;
 
+  TextEditingDeltaState? lastTextEditingDeltaState;
```

Comment:
If I understand correctly, you are using `lastTextEditingDeltaState` to accumulate all the delta information during the sequence of events (`beforeinput`, `compositionupdate`, `input`, etc) before flushing the change to the framework. If that's the case, I have a few suggestions:

1. Make the  `TextEditingDeltaState` class mutable (i.e. remove `final` from its fields) so you can update the fields in-place instead of creating many copies.
2. Remove the `last` prefix, it's confusing because it doesn't mean the same thing as `last` in `lastEditingState`.
3. Since we don't know what's the best time to create this mutable object, I suggest doing something like this:
```dart
TextEditingDeltaState? _editingDelta;
TextEditingDeltaState get editingDelta {
  if (_editingDelta == null) {
    _editingDelta = TextEditingDeltaState(oldText: lastEditingState!.text!);
  }
  return _editingDelta!;
}
```
Then you can simply do:
```dart
editingDelta.deltaStart = ...
editingDelta.deltaText = ...
```
And when the change has been flushed to the framework, just set the delta to null:
```dart
_editingDelta = null;
```

Reply:
Hey @mdebbar thank you for the reviews! I wasn't able to get to this in the past few days but I will come back to it and address all these comments. Thanks again!